pkg/registry/core/pod/storage/eviction.go

```diff
@@ -181,6 +181,11 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 
 	var rtStatus *metav1.Status
 	var pdbName string
+	updateDeletionOptions := false
+
+	v1Pod := &v1.Pod{}
+	validV1Pod := getV1Pod(pod, v1Pod)
```

Comment:
Not really a fan of this.  Our API internal endpoints should work with internal APIs.  We have `pkg/api/pod.IsPodReady` for internal pods and it is more appropriate to use than doing the conversion.

Reply:
This is a trade off.  If we want to use the identical code that PDBs use to evaluate 'healthy' then we do this.  If we don't want to use the identical code, we can use the internal pod util as you highlight.

Either option is fine for me.